### ISSUE TITLE
Add support for Swift Package Manager

### ISSUE THREAD

**Issue by carloe** [carloe/LicenseGenerator-iOS#16]
Should support SPM in addition to CocoaPods

**Comment by Schaefers** [comment]
Would be happy to see this happen!

**Comment by colejd** [comment]
You can specify the path where SPM directories are put:

```sh
./credits.py -s "$CONFIGURATION_BUILD_DIR/../../../SourcePackages/checkouts" -o "$SRCROOT/Credits.plist"
```

**Comment by gfk-sensic** [comment]
But how to combine both? Cocoapods and SPM?

**Comment by chrizstone** [comment]
But how to combine both? Cocoapods and SPM?

**Comment by svoip** [comment]
@chrizstone How did you manage it to work? Can you share? thanks

**Comment by colejd** [comment]
@chrizstone Use my PR (#38), which lets you specify multiple paths as inputs.
```
./credits.py -s "$SRCROOT, $CONFIGURATION_BUILD_DIR/../../../SourcePackages/checkouts" -o "$SRCROOT/Credits.plist"
```

* `$SRCROOT` will let it find the license files for your pods
* `$CONFIGURATION_BUILD_DIR/../../../SourcePackages/checkouts` will let it find the license files in your Swift packages. It's basically a weird way to access the DerivedData folder since I don't know a better way and there doesn't appear to be a macro for it.

**Comment by svoip** [comment]
Works great, thanks @colejd

**Comment by crspybits** [comment]
Hey-- Thanks! I've got this working with SPM now.

This is probably difficult but in one of my SPM dependencies, in its Package.swift, I make use of conditional dependencies:

```
            dependencies: [
                "ServerShared",
                // For new condition feature, see https://forums.swift.org/t/package-manager-conditional-target-dependencies/31306/26
                .product(name: "Kitura", package: "Kitura", condition: .when(platforms: [.linux, .macOS])),
                .product(name: "HeliumLogger", package: "HeliumLogger", condition: .when(platforms: [.linux, .macOS])),
                .product(name: "Credentials", package: "Kitura-Credentials", condition: .when(platforms: [.linux, .macOS])),
            ],
```

however, at least with Xcode, it looks like *all* checkout are done first (i.e., to `$CONFIGURATION_BUILD_DIR/../../../SourcePackages/checkouts`), and then the conditional rules are applied. So while my iOS app doesn't actually use some of the packages, they show up in my Acknowledgements. Not a big deal. Just makes for an inflated Acknowledgements in my Settings for the app.

Cheers!

**Comment by colejd** [comment]
Yeah, this looks to be SPM's behavior so it'd need to be changed by the Swift team. You could evaluate the Package.swift using this repo's Python script somehow, then ignore the corresponding directories that don't match, but that does sound like a lot of work.

**Comment by crspybits** [comment]
It looks like this path `$CONFIGURATION_BUILD_DIR/../../../SourcePackages/checkouts` doesn't work when archiving with Xcode. I get:

```
Showing All Messages
Error: Invalid source path: /Users/chris/Library/Developer/Xcode/DerivedData/Neebla-aybzuiaoyytaypbvqtqeummvcssh/Build/Intermediates.noindex/ArchiveIntermediates/Neebla/BuildProductsPath/Release-iphoneos/../../../SourcePackages/checkouts

Command PhaseScriptExecution failed with a nonzero exit code
```

A workaround is to put a conditional around it like:

```
if [ "${CONFIGURATION}" != "Release" ]; then
  ./credits.py -s "$SRCROOT, $CONFIGURATION_BUILD_DIR/../../../SourcePackages/checkouts" -o "$SRCROOT/Credits.plist"
fi
```

Seems a bit of a hack though. And if you really want to generate the credits on an archive, wouldn't be suitable.

**Comment by Maxismo** [comment]
> It looks like this path `$CONFIGURATION_BUILD_DIR/../../../SourcePackages/checkouts` doesn't work when archiving with Xcode. I get:
> 
> ```
> Showing All Messages
> Error: Invalid source path: /Users/chris/Library/Developer/Xcode/DerivedData/Neebla-aybzuiaoyytaypbvqtqeummvcssh/Build/Intermediates.noindex/ArchiveIntermediates/Neebla/BuildProductsPath/Release-iphoneos/../../../SourcePackages/checkouts
> 
> Command PhaseScriptExecution failed with a nonzero exit code
> ```
> 
> A workaround is to put a conditional around it like:
> 
> ```
> if [ "${CONFIGURATION}" != "Release" ]; then
>   ./credits.py -s "$SRCROOT, $CONFIGURATION_BUILD_DIR/../../../SourcePackages/checkouts" -o "$SRCROOT/Credits.plist"
> fi
> ```
> 
> Seems a bit of a hack though. And if you really want to generate the credits on an archive, wouldn't be suitable.

I just ran into this and I found the workaround for archiving.

the path is 3 levels up from there, 
`$CONFIGURATION_BUILD_DIR/../../../../../../SourcePackages/checkouts` is it.
